plugins/nf-prov/src/main/nextflow/prov/WrrocRenderer.groovy

```diff
@@ -0,0 +1,960 @@
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.SimpleFileVisitor
+import java.nio.file.StandardCopyOption
+import java.nio.file.attribute.BasicFileAttributes
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+import groovy.json.JsonOutput
+import groovy.transform.CompileStatic
+import nextflow.Session
+import nextflow.config.ConfigMap
+import nextflow.file.FileHolder
+import nextflow.processor.TaskProcessor
+import nextflow.processor.TaskRun
+import nextflow.script.params.FileInParam
+import nextflow.script.params.FileOutParam
+import nextflow.script.ScriptMeta
+import org.apache.commons.io.FilenameUtils
+import org.yaml.snakeyaml.Yaml
+
+/**
+ * Renderer for the Provenance Run RO Crate format.
+ *
+ * @author Ben Sherman <bentshermann@gmail.com>
+ * @author Felix Bartusch <felix.bartusch@uni-tuebingen.de>
+ * @author Famke Bäuerle <famke.baeuerle@qbic.uni-tuebingen.de>
+ */
+@CompileStatic
+class WrrocRenderer implements Renderer {
+
+    private Path path
+
+    private boolean overwrite
+
+    @Delegate
+    private PathNormalizer normalizer
+
+    // The final RO-Crate directory
+    private Path crateRootDir
+    // Nextflow work directory
+    private Path workdir
+    // Nextflow pipeline directory (contains main.nf, assets, etc.)
+    private Path projectDir
+    private Map agent
+    private Map organization
+    // List of contactPoints (people, organizations) to be added to ro-crate-metadata.json
+    private List<Map> contactPoints = []
+    private String publisherID
+
+    WrrocRenderer(Map opts) {
+        path = opts.file as Path
+        overwrite = opts.overwrite as Boolean
+
+        ProvHelper.checkFileOverwrite(path, overwrite)
+    }
+
+    private static final List<String> README_FILENAMES = List.of("README.md", "README.txt", "readme.md", "readme.txt", "Readme.md", "Readme.txt", "README")
+
+    @Override
+    void render(Session session, Set<TaskRun> tasks, Map<Path,Path> workflowOutputs) {
+        final params = session.params
+        final configMap = session.config
+
+        // Set RO-Crate Root and workdir
+        this.crateRootDir = Path.of(params['outdir'].toString()).toAbsolutePath()
+        this.workdir = session.getWorkDir()
+        this.projectDir = session.getWorkflowMetadata().getProjectDir()
+
+        // get workflow inputs
+        final taskLookup = ProvHelper.getTaskLookup(tasks)
+        final workflowInputs = ProvHelper.getWorkflowInputs(tasks, taskLookup)
+
+        // Add intermediate input files (produced by workflow tasks and consumed by other tasks)
+        workflowInputs.addAll(getIntermediateInputFiles(tasks, workflowInputs))
+        final workflowInputMapping = getWorkflowInputMapping(workflowInputs)
+
+        // Add intermediate output files (produced by workflow tasks and consumed by other tasks)
+        workflowOutputs.putAll(getIntermediateOutputFiles(tasks, workflowOutputs))
```

Comment:
`workflowInputs` and `workflowOutputs` are meant to contain only the pipeline inputs/outputs, not the intermediate outputs. I would keep them separate.

Reply:
done